Add airgapped/offline documentation (#2705)
canonical/ubuntu-pro-client: docs/explanations/using_pro_offline.rst

==================================================

This is an advanced use-case, and our `Customer Support teams`_ can help you
implement this in the best way for your specific environment. Note that this
requires a paid subscription to Ubuntu Pro.

What tools do I need?
=====================

Updating Ubuntu systems in an offline environment requires mirroring Ubuntu
repositories. Since Ubuntu packages can be consumed via deb packages and/or
snaps, the tool needed to mirror Ubuntu repositories depends on where you
source your packages from.

Mirror deb repositories with Landscape
--------------------------------------

Landscape is a management tool that can host a mirror of Ubuntu deb
repositories. 

In future release cycles, Landscape will also include features specific to
airgapped use-cases. You can keep up to date with these developments in the
`Landscape Beta Discourse`_.

Mirror snaps with Snap store proxy
----------------------------------

The `Snap-store-proxy`_ is the equivalent tool for hosting the snap packages that
your organisation is consuming.
 
Security patching
=================

Livepatch is the tool that provides patching for critical and high security

> vulnerabilities in the Ubuntu kernel. It can also be used in an airgapped

Author: ```suggestion
vulnerabilities in the Ubuntu kernel. It can also be used in a firewalled
```

repositories. Since Ubuntu packages can be consumed via deb packages and/or
snaps, the tool needed to mirror Ubuntu repositories depends on where you
source your packages from.

Mirror deb repositories with Landscape
--------------------------------------

Landscape is a management tool that can host a mirror of Ubuntu deb
repositories. 

In future release cycles, Landscape will also include features specific to
airgapped use-cases. You can keep up to date with these developments in the
`Landscape Beta Discourse`_.

Mirror snaps with Snap store proxy
----------------------------------

The Snap store proxy is the equivalent tool for hosting the snap packages that
your organisation is consuming.

`Snap-store-proxy`_ is a snap that provides an edge proxy to the Snap store, so
that after your device has been registered with the proxy, all communications
with the Snap store will be sent via the proxy.
 
Security patching
=================

Livepatch is the tool that provides patching for critical and high security
vulnerabilities in the Ubuntu kernel. It can also be used in an airgapped
setup using Livepatch on-prem, which our `Customer Support teams`_ can help you

> to set up.

Reviewer: I'm not sure if this part is true - I think livepatch-on-prem requires a network connection to the Canonical livepatch server. We should double check with livepatch team before promising this.

Author: I said airgapped when I think I should have said "firewalled", which is the language they use in the livepatch on-prem docs.
